api/auth.go

```diff
@@ -1,76 +1,122 @@
 package api
 
 import (
+	"crypto/ecdsa"
 	"crypto/rand"
 	"encoding/hex"
 	"errors"
-	"github.com/nuts-foundation/nuts-demo-ehr/domain/customers"
-	"github.com/sirupsen/logrus"
 	"net/http"
 	"strings"
 	"sync"
 	"time"
 
+	jwt2 "github.com/dgrijalva/jwt-go"
 	"github.com/labstack/echo/v4"
+	"github.com/lestrrat-go/jwx/jwa"
+	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/lestrrat-go/jwx/jwt/openid"
 	"github.com/nuts-foundation/nuts-demo-ehr/client"
+	"github.com/nuts-foundation/nuts-demo-ehr/domain/customers"
 )
 
 const MaxSessionAge = time.Hour
+const CustomerID = "cid"
+const SessionID = "sid"
 
 type Auth struct {
-	// sessions maps session cookies to base64 encoded VPs
-	sessions   map[string]Session
+	// sessions maps session identifiers to base64 encoded VPs
+	sessions     map[string]Session
 	// mux is used to secure access to internal state of this struct to prevent racy behaviour
-	mux        sync.Mutex
-	nodeClient client.HTTPClient
-	customers  customers.Repository
-	password   string
+	mux          sync.Mutex
+	nodeClient   client.HTTPClient
+	customers    customers.Repository
+	password     string
+	sessionKey   *ecdsa.PrivateKey
+
 }
 
 type Session struct {
 	credential interface{}
-	startTime  time.Time
 	customerID string
+	startTime  time.Time
+}
+
+type JWTCustomClaims struct {
+	CustomerID string `json:"cis"`
+	SessionID  string `json:"sid"`
+	jwt2.StandardClaims
 }
 
-func NewAuth(nodeClient client.HTTPClient, customers customers.Repository, passwd string) *Auth {
+func NewAuth(key *ecdsa.PrivateKey, nodeClient client.HTTPClient, customers customers.Repository, passwd string) *Auth {
 	return &Auth{
+		sessionKey:   key,
 		nodeClient: nodeClient,
 		customers:  customers,
 		sessions:   map[string]Session{},
 		password:   passwd,
 	}
 }
 
+// CreateCustomerJWT creates a JWT that only stores the customer ID. This is required for the IRMA flow.
+func (auth *Auth) CreateCustomerJWT(customerId string) ([]byte, error) {
+	t := openid.New()
+	t.Set(jwt.IssuedAtKey, time.Now())
+	t.Set(jwt.ExpirationKey, time.Now().Add(MaxSessionAge))
+	t.Set(CustomerID, customerId)
+
+	signed, err := jwt.Sign(t, jwa.ES256, auth.sessionKey)
```

Comment:
shouldn't we use a different key pair for this? If not for security reasons, for clarity: then you can't accidentally use the session JWT as customer JWT (the other way around probably doesn't work because it's then missing the session ID?).